### DISCUSSION TITLE
How to export dxf corresponding coordinate range to a picture based on a coordinate range

### DISCUSSION THREAD

**Discussion by asd513921** [mozman/ezdxf#1183] [Q&A] [answered]
New people want to ask, how to export dxf corresponding coordinate range into a picture according to a coordinate range, the following is my example code, always export the entire dxf file into a picture

import ezdxf
import matplotlib.pyplot as plt
from ezdxf.addons.drawing import RenderContext, Frontend
from ezdxf.addons.drawing.matplotlib import MatplotlibBackend
from ezdxf.addons.drawing.config import Configuration

# 定义坐标范围
x_min, y_min = 789970.2299576874, 12005.03740519899
x_max, y_max = 829851.4285404719, 45313.84934104128

# 读取 DXF 文件
doc = ezdxf.readfile('X-E6-incision-test.DXF')
msp = doc.modelspace()

print('读取完毕！')

# 创建 matplotlib 图形
fig, ax = plt.subplots(figsize=(200, 200))  # 设置图形大小以控制像素

print('创建 matplotlib 图形')

# 设置坐标范围
ax.set_xlim(x_min, x_max)
ax.set_ylim(y_min, y_max)
ax.set_aspect('equal', 'box')  # 确保纵横比一致

print('设置坐标范围')

# 设置渲染上下文和后端
ctx = RenderContext(doc)
out = MatplotlibBackend(ax)
config = Configuration.defaults()
config = config.with_changes(
    min_lineweight=0.1,  # in 1/300 inch: 1 mm = 1mm / 25.4 * 300
    lineweight_scaling=3,
)

print('设置渲染上下文和后端')

# 使用 Frontend 将模型空间渲染到 matplotlib 图形
frontend = Frontend(ctx, out, config)
print('使用 Frontend 绘制中')
frontend.draw_layout(msp)  # 调整 lineweight 来控制线条的粗细

print('使用 Frontend 将模型空间渲染到 matplotlib 图形')

# 保存为图片
fig.savefig('output.png', dpi=1000, bbox_inches='tight')  # 调整 dpi 和 bbox_inches
plt.close(fig)

print("图片保存成功！")

**Comment by mozman** [comment, marked as answer]
see FAQ: #550

- DXF (WCS) to image transform: #219
- image to DXF (WCS) transform: #269
